packages/docgen/test/get-type-annotation.js

```diff
@@ -263,4 +264,36 @@ describe( 'Type annotations', () => {
 			).toBe( '( foo: string, ...rest: any[] ) => GenericType< T >' );
 		} );
 	} );
+
+	describe( 'function argument object destructuring', () => {
+		describe( 'type literal', () => {
+			const node = getObjectDestructuringTypeLiteralNode();
+
+			it( 'should get the full type for the param', () => {
+				expect( getTypeAnnotation( paramTag, node, 0 ) ).toBe(
+					'{ foo: string; }'
+				);
+			} );
+
+			it( 'should get the member type for the param', () => {
+				expect(
+					getTypeAnnotation(
+						{ ...paramTag, name: 'props.foo' },
+						node,
+						0
+					)
+				).toBe( 'string' );
+			} );
+
+			it( 'should get the member type for an unfindable member', () => {
+				expect(
+					getTypeAnnotation(
+						{ ...paramTag, name: 'props.notFoo' },
+						node,
+						0
+					)
+				).toBe( "{ foo: string; }[ 'notFoo' ]" );
```

Comment:
🤔 I'm not a huge fan of this, but what do you think? I think I might expect it to just be `unknown` -- is that even a thing in docgen?

Reply:
Well, this is technically an edge case that wouldn't compile:

```ts
function fn({ notFoo }: {foo: string}) //...
```

Would throw a compilation error.

The main use case for this kind of indexed access is something like a type reference:

```ts
function fn({ notFoo }: SomeType) //...
```

In which case we'd get `SomeType[ 'notFoo' ]` which I think is decent.

Maybe in the case where something is an object literal and we can't find the member type we should just return `unknown` even though that's not a valid TypeScript program to begin with?

That sort of begs the question of whether we should be in the business of handling invalid programs, I'm not super sure.

Reply:
makes sense

> we should be in the business of handling invalid programs

Yeah... I think my inclination is no. It's probably fair to expect compile-able Typescript as an input